cid-decode/src/main.rs

```diff
@@ -53,51 +67,163 @@ fn main() -> Result<()> {
         let entry = results.entry(res.clone()).or_default();
         *entry += 1;
 
-        s.clear();
+        buffer.clear();
     }
     // Restore panic hook.
-    panic::set_hook(h);
+    panic::set_hook(default_panic_hook);
```

Comment:
Can you tell me where I can get a few million CIDs?

Reply:
I also think its probably safe to remove
Why was that added in the first place?

Reply:
I added this because one of the `cid`, `multihash`, ... , crates would panic instead of return an error :upside_down_face: 
I think we should be pretty safe from panics now, but I'd like to have it tested with some sample of real CIDs.
Did you get around to that yet? If it doesn't panic and produces reasonable results, feel free to remove all the panic handling.